Add X-VA-User header to appeals status service api

edpaget opened the pull request:
Resolves department-of-veterans-affairs/vets-contrib#169

We are passing a 'consumer' and 'va-user' header onto caseflow.

* The 'consumer' header is the username assigned to the api key in Kong and represents the application making the request
* The 'va-user' header is the username of the person making the request to caseflow. It will be logged to CorpDB by caseflow.

patrickvinograd commented:
Can you clarify in the PR description:
- What header names we are passing to caseflow.
- Where each of them is derived from.

I think we're overloading "user" to mean a lot of different things - looks like we are now passing "Consumer-Username" through to appeals but that means the API consumer which is probably an application name? And then the code has a `user` method that gives the target user, plus the `requesting_user`. If we want to rename any of these headers at any tier better to do it now.

edpaget commented:
I added more description of the names of the headers above. I also updated variable names to hopefully clarify that the requesting user, target veteran, and consumer are all different. Can also update those further if it makes the controller more understandable.

patrickvinograd commented on modules/appeals_api/app/controllers/appeals_api/v0/appeals_controller.rb:
```diff
@@ -11,7 +11,8 @@ def index
         appeals_response = Appeals::Service.new.get_appeals(
           user,
-          'X-Consumer-Username' => request.headers['X-Consumer-Username']
+          'Consumer-Username' => consumer
```
Needs a comma
